crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4180,6 +4224,104 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn except_handler() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            "
+            import builtins
+
+            try:
+                x
+            except NameError as e:
+                pass
+            except (TypeError, builtins.AttributeError) as f:
```

Comment:
This is more a commentary on testing philosophy than a request for you to change this, but the way I generally think about tests is to aim for each test to be clear about what it's testing, and to be minimalist in every other aspect. This leads to more focused test failures (rather than one breakage causing many seemingly unrelated tests to fail), and tests whose purpose are easier for a reader to understand.

So I would say, we have tests elsewhere (at least I think we do!) that builtins can be accessed either implicitly or via `import builtins`, and nothing in our exception-handler code does or should do anything special regarding this (it's just normal expression type resolution.) So I don't see much reason why the wrinkles of `import builtins` or using an Attribute node as an exception type belong in this test at all. It just makes the reader wonder why they are there, and if they are missing something about their importance.

Of course this philosophy has its limits; some very basic syntactic forms are of course relied on by many tests, because you need them in order to even use the feature you want to test. But both `import builtins` and exception handling are rather specific, less common features, so I would generally not have them in the same test together unless I am intentionally testing some very particular interaction of precisely those two things. And in that case, it would be in a very specific separate test whose name clearly indicates what unusual interaction we are testing, it wouldn't be in a generic `except_handler` test.

Reply:
Thanks for the great comment. I did want to include an exception class from another module _somewhere_ in my tests -- it probably isn't _necessary_ since we obviously demonstrate support for cross-module resolution elsewhere, but it felt useful to show that we'd infer the types even across module boundaries. But you're absolutely right that the separation of my tests wasn't very clean! I've cleaned them up somewhat in https://github.com/astral-sh/ruff/pull/13267/commits/73eb467237a66aa2f79f314015405bb37869fa76